src/Http/Http/perf/Microbenchmarks/QueryCollectionBenchmarks.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using BenchmarkDotNet.Attributes;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Primitives;
+using static Microsoft.AspNetCore.Http.Features.QueryFeature;
+
+namespace Microsoft.AspNetCore.Http
+{
+    public class QueryCollectionBenchmarks
+    {
+        private string _queryString;
+        private string _singleValue;
+
+        [IterationSetup]
+        public void Setup()
+        {
+            _queryString = "?key1=value1&key2=value2&key3=value3&key4=&key5=";
+            _singleValue = "?key1=value1";
```

Comment:
These can be assigned to the fields directly. Just don't make the fields const or static readonly.

Reply:
Should add a case where some `+` are contained, to reflect the allocations for the string.Replace?